src/yunohost/backup.py

```diff
@@ -199,257 +300,374 @@ def _clean_tmp_dir(retcode=0):
-    backup_size = int(subprocess.check_output(
-        ['du', '-sb', tmp_dir]).split()[0].decode('utf-8'))
-    info['size'] = backup_size
-
-    # Create backup info file
-    with open("%s/info.json" % tmp_dir, 'w') as f:
-        f.write(json.dumps(info))
-
-    # Create the archive
-    if not no_compress:
-        logger.info(m18n.n('backup_creating_archive'))
-
-        # Check free space in output directory at first
-        avail_output = subprocess.check_output(
-            ['df', '--block-size=1', '--output=avail', tmp_dir]).split()
+                return 0
+        self.info['size'] = self._call_for_each_path(_compute_path_size)
+        return self.info['size']
+
+    def _check_is_enough_free_space(self, output_directory):
+        """ Check free space in output directory at first """
+        backup_size = self.info['size']
+        cmd = ['df', '--block-size=1', '--output=avail', output_directory]
+        avail_output = subprocess.check_output(cmd).split()
         if len(avail_output) < 2 or int(avail_output[1]) < backup_size:
             logger.debug('not enough space at %s (free: %s / needed: %d)',
                          output_directory, avail_output[1], backup_size)
-            _clean_tmp_dir(3)
+            self.clean(3)
             raise MoulinetteError(errno.EIO, m18n.n(
                 'not_enough_disk_space', path=output_directory))
 
+    def _copy_files(self, output_directory):
+        """ Copy prepared files into a dir """
+        def _copy_path(self, source, dest):
+            dest = os.path.join(self.collect_dir, dest)
+            if source == dest:
+                return
+
+            dest_parent = os.path.dirname(dest)
+            if not os.path.exists(dest_parent):
+                filesystem.mkdir(dest_parent, 0750, True, uid='admin')
+
+            if os.path.isdir(source):
+                shutil.copytree(source, dest)
+            else:
+                shutil.copy(source, dest)
+        # Check free space in output
+        self._check_is_enough_free_space(output_directory)
+
+        self._call_for_each_path(_copy_path)
+
+    def _tar_files(self, output_directory):
+        """ Compress prepared files """
+        # Check free space in output
+        self._check_is_enough_free_space(output_directory)
+
         # Open archive file for writing
-        archive_file = "%s/%s.tar.gz" % (output_directory, name)
+        archive_file = os.path.join(output_directory, self.name + '.tar.gz')
         try:
             tar = tarfile.open(archive_file, "w:gz")
         except:
             logger.debug("unable to open '%s' for writing",
                          archive_file, exc_info=1)
-            _clean_tmp_dir(2)
+            self.clean(2)
             raise MoulinetteError(errno.EIO,
                                   m18n.n('backup_archive_open_failed'))
 
         # Add files to the archive
+        def _tar_path(self, source, dest):
+            tar.add(source, arcname=dest)
```

Comment:
@JocelynDelalande  It can't be explicit, tar.add is define by the tarfile python library, not by yunohost.
Or may be you want to rename dest into archive_name ?